ocaml/xapi/xapi_host.ml

```diff
@@ -1124,10 +1124,52 @@ let crl_list ~__context ~host =
   Certificates.(local_list CRL)
 
 let certificate_sync ~__context ~host =
-  Certificates.local_sync()
+  Certificates.local_sync ()
 
 let get_server_certificate ~__context ~host =
-  Certificates.get_server_certificate()
+  Certificates.get_server_certificate ()
+
+let install_server_certificate ~__context ~host ~certificate ~private_key ~certificate_chain =
+  if Db.Pool.get_ha_enabled ~__context ~self:(Helpers.get_pool ~__context) then
+    raise Api_errors.(Server_error (ha_is_enabled, []));
+
+  let current_server_certs = Db.Certificate.get_all_records ~__context
+  |> List.filter_map (fun (reference, record) ->
+    if record.API.certificate_host = host then
+        Some reference
+    else
+        None
+    )
+  in
+
+  let pem_chain = match certificate_chain with
```

Comment:
worth trimming here? :man_shrugging:

Reply:
pem_chain is an "optional" parameter, `no chain` is represented by the empty string